src/components/QueryRecordings/QueryRecordings.vue

```diff
@@ -222,18 +236,32 @@ export default {
     devicesDescription() {
       const numDevices = this.selectedDevices.length;
       const numGroups = this.selectedGroups.length;
-      const total = numDevices + numGroups;
+      const numStations = this.selectedStations.length;
+      const total = numDevices + numGroups + numStations;
 
       const multipleSuffix = total > 1 ? "s" : "";
 
       if (total === 0) {
         return "All devices";
-      } else if (numDevices && numGroups) {
-        return `${total} groups and devices`;
-      } else if (numGroups) {
+      } else if (numGroups && !numDevices && !numStations) {
         return `${total} group${multipleSuffix}`;
+      } else if (numStations && !numDevices && !numGroups) {
+        return `${total} station${multipleSuffix}`;
+      } else if (numDevices && !numStations && !numGroups) {
+        return `${total} device${multipleSuffix}`;
+      } else {
+        const items = [];
+        if (numGroups) {
+          items.push("groups");
+        }
+        if (numDevices) {
+          items.push("devices");
+        }
+        if (numStations) {
+          items.push("stations");
+        }
+        return `${total} ${items.join(" and ")}`;
```

Comment:
This ideally would be `groups, devices and stations` if all 3 exists rather than `groups and devices and stations`